src/core/execute.h

```diff
@@ -120,6 +122,9 @@ struct ExecContext {
         ExecInput std_input;
         ExecOutput std_output;
         ExecOutput std_error;
+        char *std_input_fdname;
+        char *std_output_fdname;
+        char *std_error_fdname;
```

Comment:
hmm, I think we could simplify this a lot, if we'd change this to become an array. i.e.:

```c
char *stdio_fdname[3];
```

And then we could access the three members as `c->stdio_fdname[STDIN_FILENO]`, c->stdio_fdname[STDOUT_FILENO]`, …, you get the idea... I am pretty sure a lot of code accessing these names would become a lot simpler through this as well.